sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -4100,13 +4099,8 @@ object functions {
    * @group collection_funcs
    * @since 2.3.0
    */
-  def from_json(e: Column, schema: String, options: Map[String, String]): Column = {
-    val dataType = parseTypeWithFallback(
-      schema,
-      DataType.fromJson,
-      "Cannot parse the schema in JSON format: ",
-      fallbackParser = DataType.fromDDL)
-    from_json(e, dataType, options)
+  def from_json(e: Column, schema: String, options: Map[String, String]): Column = withExpr {
+    new JsonToStructs(e.expr, lit(schema).expr, options)
```

Comment:
Wait .. I remember actually I had a discussion about this (with @maropu ..?) before. JSON format schema is actually pretty internal ser/de purpose, and just encouraging DDL format is better, and JSON format here was kept just for the compatibility reason.

Reply:
Okay but `from_json` itself supports anyway. Okay should be best to match across other languages for the same API.